Packs/ApiModules/Scripts/AggregatedCommandApiModule/AggregatedCommandApiModule.py

```diff
@@ -136,6 +138,75 @@ def get_all_values_from(self, data: dict[str, Any]) -> dict[str, Any]:
         return out
 
 
+@dataclass
+class IndicatorInstance:
+    """
+    Represents a single indicator undergoing the full enrichment pipeline.
+
+    This object tracks the state of one input value across all stages of the
+    enrichment flow (extract → create → enrich → TIM → final context).
+    It centralizes the status, errors, enrichment results, and final outcome,
+    making it easier to build the final context entries and human-readable output.
+
+    Attributes:
+        raw_input (str): The original input string provided by the user.
+        extracted_value (str | None): The indicator value extracted by `extractIndicators`.
+            None if extraction failed or returned no value of the expected type.
+        created (bool): Whether the indicator was successfully created by `CreateNewIndicatorsOnly`.
+        enriched (bool): Whether external enrichment (`enrichIndicators` or internal commands) succeeded for this indicator.
+        tim_context (list[ContextResult] | None): List of TIM results (standardized enrichment results) returned from TIM search.
+            None if no TIM results were found.
+        hr_message (str | None): Message for human readable.
+        error_message (str | None): Accumulated error message from enrichment pipeline.
+        context_message (str | None): Small message for Context, relevant for failures.
+        final_status (Status): The computed status of the indicator after all enrichment stages.
+            Typically SUCCESS unless any stage failed or an invalid input was detected.
+    """
+
+    raw_input: str
+    extracted_value: str | None = None
+    created: bool = False
+    enriched: bool = False
+    tim_context: list[ContextResult] | None = None
+    hr_message: str | None = None
+    context_message: str | None = None
+    error_message: str | None = None
+    final_status: Status = Status.SUCCESS
+
+    def compute_status(self) -> None:
```

Comment:
Let's split this function into smaller helper methods (e.g., is_valid(), tim_found(), build_failure_context()).
```

    def _is_valid(self) -> bool:
        return bool(self.extracted_value)

    def _tim_found(self) -> bool:
        return bool(self.tim_context)

    def _build_failure_context(self) -> str | None:
        """Builds a consistent failure context message."""
        messages: list[str] = []
```